### DISCUSSION TITLE
Backup fails after a restore

### DISCUSSION THREAD

**Discussion by mrickl** [abbbi/virtnbdbackup#48] [Q&A / Support] [answered]
First of all, thank you for the great project here. It even helped me a lot yesterday when I had to restore one of my VMs as "Point in time recovery". But I can't make any "incremental" backups now, because I had to fall back on the second incremental backup, for example. But as soon as I want to make a new incremental backup, it is noted that the checkpoint already exists. Can I simply delete everything that occurs after the restored checkpoint from my backup directory and from the external checkpoint directory? Or is it smart to delete everything and create a new full backup?

best regards

Marcel

**Comment by abbbi** [comment, marked as answer]
Restoring a virtual machine will create a new qcow image. The bitmap information is saved in two places:

1) libvirt saves checkpoint information
2) the qcow images include the dirty bitmap tied to the checkpoint.

If you have restored the virtual machine disk, there are no bitmaps existant in the restored qcow image.
You have to create a new full backup chain, but virtnbdbackup will fail to remove the checkpoints because
of the missing bitmap.

To cleanup the situation remove the existing checkpoints from libvirt using:

virsh checkpoint-list [..]
virsh checkpoint-delete [..]

if virsh checkpoint-delete fails because it is missing the bitmaps (likely), remove the checkpoint metadata only using
the 

--metadata [..]

option. Then start a new full backup to a new directory.

**Comment by mrickl** [comment]
@abbbi thanks for the information, that worked for me with "--metadata". Wouldn't that be something for the FAQ or as info text under Restore? Then the next ones, who also have the problem, could find it faster if they face the same problem.
